plugins/partiql-memory/src/main/kotlin/org/partiql/plugins/memory/MemoryBindings.kt

```diff
@@ -0,0 +1,52 @@
+package org.partiql.plugins.memory
+
+import com.amazon.ionelement.api.IonElement
+import com.amazon.ionelement.api.StructElement
+import org.partiql.spi.connector.ConnectorBindings
+import org.partiql.spi.connector.ConnectorObjectPath
+import org.partiql.value.PartiQLValue
+import org.partiql.value.PartiQLValueExperimental
+import org.partiql.value.io.PartiQLValueIonReaderBuilder
+import org.partiql.value.missingValue
+
+@OptIn(PartiQLValueExperimental::class)
+class MemoryBindings(
+    private val bindings: Map<String, PartiQLValue>,
+) : ConnectorBindings {
+
+    @OptIn(PartiQLValueExperimental::class)
+    override fun getValue(path: ConnectorObjectPath): PartiQLValue {
+        val key = path.steps.joinToString(".")
+        return bindings[key] ?: missingValue()
+    }
+
+    internal fun iterator(): Iterator<PartiQLValue> = bindings.values.iterator()
+
+    companion object {
+
+        /**
+         * No bindings.
+         */
+        val empty = MemoryBindings(emptyMap())
+
+        /**
+         * Loads each declared global of the catalog from the data element.
+         */
+        fun load(metadata: MemoryConnector.Metadata, data: StructElement): MemoryBindings {
```

Comment:
```suggestion
        fun load(metadata: MemoryConnector.Metadata, data: Map<String, PartiQLValue>): MemoryBindings {
```
While this is used for the conformance tests, I think this logic would more adequately fit there. The current implementation is specific to the conformance tests.

Reply:
Makes sense. I moved `load` to be a private function within the conformance test runner.